pkg/registry/common/expire/nse_server.go

```diff
@@ -51,102 +53,72 @@ func NewNetworkServiceEndpointRegistryServer(ctx context.Context, nseExpiration
 }
 
 func (n *expireNSEServer) Register(ctx context.Context, nse *registry.NetworkServiceEndpoint) (*registry.NetworkServiceEndpoint, error) {
-	t, loaded := n.timers.LoadAndDelete(nse.Name)
+	t, loaded := n.timers.Load(nse.Name)
+	// TODO: this is totally incorrect if there are concurrent events for the same nse.Name, think about adding
+	//       serialize into the registry chain
```

Comment:
Should we consider this at first? What is the issue number?

Reply:
Yes, probably : https://github.com/networkservicemesh/sdk/issues/760.